common/service-support-cdi/src/main/java/io/helidon/common/servicesupport/cdi/CdiExtensionBase.java

```diff
@@ -0,0 +1,483 @@
+    }
+
+    /**
+     * Cleans up any data structures created during annotation processing but which are not needed once the CDI container has
+     * started.
+     *
+     * @param adv the {@code AfterDeploymentValidation} event
+     */
+    protected void clearAnnotationInfo(@Observes AfterDeploymentValidation adv) {
+        if (logger.isLoggable(Level.FINE)) {
+            Set<Class<?>> annotatedClassesIgnored = new HashSet<>(annotatedClasses());
+            annotatedClassesIgnored.removeAll(annotatedClassesProcessed());
+            if (!annotatedClassesIgnored.isEmpty()) {
+                logger.log(Level.FINE, () ->
+                        "Classes originally found with selected annotations that were not processed, probably "
+                                + "because they were vetoed:" + annotatedClassesIgnored.toString());
+            }
+        }
+        annotatedClasses.clear();
+        annotatedClassesProcessed.clear();
+    }
+
+    protected R restEndpointInfo() {
+        return restEndpointInfo;
+    }
+
+    /**
+     * Finds an existing or adds a new extension-specific {@code AsyncResponseInfo} for the indicated method.
+     *
+     * @param method the Method for which the AsyncResponseInfo is needed
+     * @return the pre-existing or newly-created instance, if any; null if no existing mapping was found and the factory method
+     * declined to create one
+     */
+    protected A computeIfAbsentAsyncResponseInfo(Method method){
+        Map<Method, A> asyncResponseInfo = restEndpointInfo.asyncResponseInfo();
+        return asyncResponseInfo.computeIfAbsent(method, this::newAsyncResponseInfo);
+    }
+
+    /**
+     * Returns a {@code AsyncResponseInfo} (or subclass) instance describing the async information about the specified method.
+     *
+     * @param method Method to examine for asynchronous behavior
+     * @return AsyncResponseInfo describing the async behavior; null if the method is synchronous
+     */
+    protected abstract A newAsyncResponseInfo(Method method);
+
+    /**
+     * Returns the index in the method's array of parameters, if any, with type {@code AsyncResponse} and annotated with
+     * {@code Suspended}.
+     *
+     * @param m the method to examine
+     * @return the array index of the async parameter, if any; -1 otherwise
+     */
+    protected static int asyncParameterSlot(Method m) {
+        int candidateAsyncResponseParameterSlot = 0;
+
+        for (Parameter p : m.getParameters()) {
+            if (AsyncResponse.class.isAssignableFrom(p.getType()) && p.isAnnotationPresent(Suspended.class)) {
+                return candidateAsyncResponseParameterSlot;
+            }
+            candidateAsyncResponseParameterSlot++;
+
+        }
+        return -1;
+    }
+
+    /**
+     * Returns a new instance of the extension-specific REST endpoint information.
+     *
+     * @return newly-initialized instance
+     */
+    protected abstract R newRestEndpointInfo();
```

Comment:
Not all extensions have rest endpoints. I am not sure this method belongs in `common` module, if it is not for any extension, just for the ones that expose an endpoint.

Reply:
Also the javadoc does not really explain what this is for

Reply:
Addressed by renaming/refactoring as described here: https://gist.github.com/tjquinno/d5b5d13d4e1220e6631372c59d2f7d8b